### PR TITLE
Move validation code to its own package

### PR DESCRIPTION
**Pull request by bbguimaraes**
The single file is getting very dense.  Moving to separate files in a separate
package, with a small simplification on top.

### CODE REVIEW

**Comment by bbguimaraes** [comment]
https://issues.redhat.com/browse/DPTP-1365

File: cmd/ci-operator/main.go [hunk 1]
```diff
@@ -422,7 +423,7 @@ func (o *options) Complete() error {
 		o.jobSpec.Refs.PathAlias = *config.CanonicalGoRepository
 	}
 	o.configSpec = config
-	if err := o.configSpec.ValidateResolved(); err != nil {
+	if err := validation.ValidateResolvedConfiguration(o.configSpec); err != nil {
```
Comment (petr-muller) [inline cmd/ci-operator/main.go:426]: if we called the package `validate` then methods would not need to contain `Validate` and this would not stutter.

Comment (bbguimaraes) [reply cmd/ci-operator/main.go]: Changed public functions to use the same pattern as `k8s.io/apimachinery/pkg/util/validation`, which is where the package name came from.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**